examples/notebooks/petsc_eigenvalues_regularized_curlcurl.ipynb

```diff
@@ -0,0 +1,282 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "2443e734",
+   "metadata": {},
+   "source": [
+    "# Example of computing approximative eigenvalues with PETSc\n",
+    "\n",
+    "In this notebook we show how to compute approximative eigenvalues using PETSc. We use Conjugate Gradient to solve a 3D problem in parallel. Besides, we use a Jacobi preconditioner to illustrate how to set up a preconditioned Krylov solver with PETSc. The eigenvalues are estimated using the Krylov method and the number of estimated eigenvalues is the number of iterations.\n",
+    "\n",
+    "Similar to the 3D VTK ParaView test, we solve the regularized curl curl problem in 3D with a callable function for the source. In practice, the source could be the interpolation of some experimental data. Besides, we impose periodic boundary conditions everywhere in the boundary.\n",
+    "\n",
+    "Let $\\Omega=[0, 10] \\times [-2\\pi, 2\\pi]\\times [-3, 1]$. We want to find $u\\in H(\\mathrm{curl}, \\Omega)$ such that\n",
+    "\n",
+    "$$ \\mathrm{curl} \\mathrm{curl} u + \\mu u = f, $$\n",
+    "\n",
+    "where $\\mu = \\pi$, and $f$ is a callable function in terms of the domain coordinates, the expression of which is a priori not known. \n",
+    "\n",
+    "Only for illustration here we use the callable function\n",
+    "$$f(x,y,z) = (10e^{-0.5(x-4)^2 - 1.5z^2 - 0.1y^2}, 0, \\cos(y)\\sin(\\pi x / 5)),$$\n",
+    "\n",
+    "and project it from $H(\\mathrm{curl}, \\Omega)$ to the corresponding discretized space. This process is analogous with any callable vectorial function.\n",
+    "\n",
+    "\n",
+    "\n",
+    "## Step 1 : define the domain and spaces and discretize them\n",
+    "\n",
+    "This is similar to what is done in other examples"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "39c21151",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "from mpi4py                     import MPI\n",
+    "from sympde.topology            import Cube\n",
+    "from sympde.topology            import VectorFunctionSpace\n",
+    "from psydac.api.discretization  import discretize\n",
+    "\n",
+    "comm = MPI.COMM_WORLD # MPI communicator\n",
+    "\n",
+    "Omega = Cube('Omega', bounds1=(0, 10), bounds2=(-2*np.pi, 2*np.pi), bounds3=(-3, 1)) # 3D cubic domain\n",
+    "\n",
+    "W = VectorFunctionSpace('W', Omega, kind='hcurl') # Hcurl space\n",
+    "\n",
+    "ncells = [25, 30, 11] # number of cells in each direction\n",
```

Comment:
Should we decrease the number of cells here? The CI takes a long time to run already and this notebook takes most of the time.

Reply:
Changing the backend to pyccel did the trick as well.